### ISSUE TITLE
[UnnecessaryLet] False positive when it is original variable is used instead of it

### ISSUE THREAD

**Issue by atulgpt** [detekt/detekt#7767]
## Expected Behavior
Both of the below TC should pass

```kotlin
@Test
fun `when let uses the it`() {
    val content = """
        fun some(value: Int) = value
        fun test(value: Int?) {
            value?.let {
                some(it)
            }
        }
    """.trimIndent()
    val findings = subject.compileAndLintWithContext(env, content)
    assertThat(findings).hasSize(1)
}

@Test
fun `when let uses the variable`() {
    val content = """
        fun some(value: Int) = value
        fun test(value: Int?) {
            value?.let {
                some(value)
            }
        }
    """.trimIndent()
    val findings = subject.compileAndLintWithContext(env, content)
    assertThat(findings).hasSize(1)
}
```

## Observed Behavior
TC `when let uses the it` is passing while TC is `when let uses the variable` is failing. 
If we look into the code there is no diff between in one case we are using `it` while in other we are using `value`

## Context
I am not sure what is the role of `UnnecessaryLet` here is this to promote `if` or just avoid case when `?.let` can be simplified with the safe operator(`?.`)

## Steps to Reproduce
Run the TC in the latest detekt project

**Comment by atulgpt** [comment]
Not stale

**Comment by t-kameyama** [comment]
I believe the following tests should pass:

```kotlin
@Test
fun `when let uses the it`() {
    val content = """
        fun some(value: Int) = value
        fun test(value: Int?) {
            value?.let {
                some(it)
            }
        }
    """.trimIndent()
    val findings = subject.lintWithContext(env, content)
    assertThat(findings).isEmpty()
}

@Test
fun `when let uses the variable`() {
    val content = """
        fun some(value: Int) = value
        fun test(value: Int?) {
            value?.let {
                some(value)
            }
        }
    """.trimIndent()
    val findings = subject.lintWithContext(env, content)
    assertThat(findings).isEmpty()
}
```

The `when let uses the variable` is a false positive.

**Comment by BraisGabin** [comment]
These cars could be replaced by an `if`. Wouldn't it be a better option? 

The problem of this rule is where you draw the line. I started a lot of times to describe what is and what is not an unnecessary use of let but it's really difficult to define. There are clear cases but a ton of grey ones.

**Comment by t-kameyama** [comment]
True, we could use an `if` here, but I wouldn't call this let "unnecessary." It's hard to define the line, but I think the current implementation is acceptable.
